Failing unit test: TestModelFormConfiguration.test_supports_custom_datetime_format

With a fresh clone of the repo and running `tox -e sqlalchemy14`, one of the unit tests fails. This is the output after installing dependencies:
```
================================================= test session starts =================================================
platform win32 -- Python 3.11.4, pytest-7.4.0, pluggy-1.2.0
cachedir: .tox\sqlalchemy14\.pytest_cache
rootdir: C:\Users\Vidminas\GitHub\wtforms-alchemy
plugins: anyio-3.7.1, cov-4.1.0
collected 248 items

tests\test_class_map.py .................                                                                        [  6%]
tests\test_column_aliases.py .....                                                                               [  8%]
tests\test_configuration.py ...........F......                                                                   [ 16%]
tests\test_country_field.py ..                                                                                   [ 16%]
tests\test_custom_fields.py .                                                                                    [ 17%]
tests\test_deep_form_relations.py ..                                                                             [ 18%]
tests\test_descriptions.py ..                                                                                    [ 18%]
tests\test_field_exclusion.py ....                                                                               [ 20%]
tests\test_field_order.py .                                                                                      [ 20%]
tests\test_field_parameters.py .............                                                                     [ 26%]
tests\test_field_trimming.py ..                                                                                  [ 27%]
tests\test_form_meta.py ........                                                                                 [ 30%]
tests\test_hybrid_properties.py ..                                                                               [ 31%]
tests\test_i18n_extension.py ...                                                                                 [ 32%]
tests\test_inheritance.py .....                                                                                  [ 34%]
tests\test_labels.py ..                                                                                          [ 35%]
tests\test_model_field_list.py ........                                                                          [ 38%]
tests\test_model_form_factory.py ............                                                                    [ 43%]
tests\test_model_form_field.py ...                                                                               [ 44%]
tests\test_phone_number.py ....                                                                                  [ 45%]
tests\test_phone_number_field.py ............                                                                    [ 50%]
tests\test_query_select_field.py .............                                                                   [ 56%]
tests\test_select_field.py ..........                                                                            [ 60%]
tests\test_synonym.py ..                                                                                         [ 60%]
tests\test_types.py ..............................................                                               [ 79%]
tests\test_unique_validator.py .......................                                                           [ 88%]
tests\test_utils.py .                                                                                            [ 89%]
tests\test_validators.py ...................                                                                     [ 96%]
tests\test_weekdays_field.py ..                                                                                  [ 97%]
tests\test_widgets.py ......                                                                                     [100%]

====================================================== FAILURES =======================================================
___________________________ TestModelFormConfiguration.test_supports_custom_datetime_format ___________________________

self = <tests.test_configuration.TestModelFormConfiguration object at 0x000001883DB515D0>

    def test_supports_custom_datetime_format(self):
        self.init(sa.DateTime, nullable=False)

        class ModelTestForm(ModelForm):
            class Meta:
                model = self.ModelTest
                datetime_format = '%Y-%m-%dT%H:%M:%S'

        form = ModelTestForm()
>       assert form.test_column.format == '%Y-%m-%dT%H:%M:%S'
E       AssertionError: assert ['%Y-%m-%dT%H:%M:%S'] == '%Y-%m-%dT%H:%M:%S'
E        +  where ['%Y-%m-%dT%H:%M:%S'] = <wtforms_components.fields.html5.DateTimeField object at 0x000001883DF97110>.format
E        +    where <wtforms_components.fields.html5.DateTimeField object at 0x000001883DF97110> = <tests.test_configuration.TestModelFormConfiguration.test_supports_custom_datetime_format.<locals>.ModelTestForm object at 0x000001883DEBB190>.test_column

tests\test_configuration.py:154: AssertionError
================================================== warnings summary ===================================================
tests\test_unique_validator.py:12
  C:\Users\Vidminas\GitHub\wtforms-alchemy\tests\test_unique_validator.py:12: MovedIn20Warning: Deprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. To prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to "sqlalchemy<2.0". Set environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message. (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
    base = declarative_base()

tests/test_model_form_factory.py::TestModelFormFactory::test_class_meta_wtforms2
tests/test_model_form_factory.py::TestModelFormFactory::test_class_meta_wtforms2
  C:\Users\Vidminas\GitHub\wtforms-alchemy\tests\test_model_form_factory.py:94: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(wtforms.__version__) < LooseVersion('2'):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================== short test summary info ===============================================
FAILED tests/test_configuration.py::TestModelFormConfiguration::test_supports_custom_datetime_format - AssertionError...
====================================== 1 failed, 247 passed, 3 warnings in 2.67s ======================================
sqlalchemy14: exit 1 (5.03 seconds) C:\Users\Vidminas\GitHub\wtforms-alchemy> py.test pid=9716
.pkg: _exit> python C:\Users\Vidminas\.conda\envs\wtforms-alchemy\Lib\site-packages\pyproject_api\_backend.py True setuptools.build_meta __legacy__
  sqlalchemy14: FAIL code 1 (92.41=setup[49.20]+cmd[38.17,5.03] seconds)
  evaluation failed :( (92.62 seconds)
```

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -2,13 +2,14 @@ name: Tests
 on: [push, pull_request]
 jobs:
   test:
-    name: Python ${{ matrix.python-version }} + SQLAlchemy ${{ matrix.sqlalchemy-version }}
+    name: Python ${{ matrix.python-version }} + SQLAlchemy ${{ matrix.sqlalchemy-version }} + WTForms ${{ matrix.wtforms-version }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
         sqlalchemy-version: ["1.4", "2.0"]
+        wtforms-version: ["3.1", "3.2"]
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python
@@ -18,4 +19,4 @@ jobs:
       - name: Install tox
         run: pip install tox
       - name: Run tests
-        run: tox -e sqlalchemy${{ matrix.sqlalchemy-version }}
+        run: tox -e sqlalchemy${{ matrix.sqlalchemy-version }}-wtforms${{ matrix.wtforms-version }}

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -10,6 +10,7 @@ Unreleased
 - Dropped support for SQLAlchemy 1.3 and earlier. The minimum supported SQLAlchemy version is now 1.4.
 - Added support for Python 3.10–3.13.
 - Added support for SQLAlchemy 2.0.
+- Added support for WTForms 3.2. The minimum supported WTForms version is now 3.1.
 
 0.18.0 (2021-12-21)
 ^^^^^^^^^^^^^^^^^^^

--- a/setup.py
+++ b/setup.py
@@ -69,8 +69,8 @@ setup(
     platforms='any',
     install_requires=[
         'SQLAlchemy>=1.4',
-        'WTForms>=2.2,<=3.0.0',
-        'WTForms-Components>=0.9.2',
+        'WTForms>=3.1.0',
+        'WTForms-Components>=0.11.0',
         'SQLAlchemy-Utils>=0.32.6'
     ],
     extras_require=extras_require,

--- a/tests/test_configuration.py
+++ b/tests/test_configuration.py
@@ -151,7 +151,7 @@ class TestModelFormConfiguration(ModelFormTestCase):
                 datetime_format = '%Y-%m-%dT%H:%M:%S'
 
         form = ModelTestForm()
-        assert form.test_column.format == '%Y-%m-%dT%H:%M:%S'
+        assert form.test_column.format == ['%Y-%m-%dT%H:%M:%S']
 
     def test_supports_additional_validators(self):
         self.init()

--- a/tests/test_field_parameters.py
+++ b/tests/test_field_parameters.py
@@ -81,6 +81,7 @@ class TestFieldParameters(ModelFormTestCase):
     def test_uses_custom_field_class(self):
         class InputTest(widgets.Input):
             input_type = 'color'
+            validation_attrs = []
 
         class FieldTest(StringField):
             widget = InputTest()
@@ -102,12 +103,6 @@ class TestFieldParameters(ModelFormTestCase):
         assert 'type="color"' in str(form.test_column)
 
     def test_accepts_none_as_custom_field_class(self):
-        class InputTest(widgets.Input):
-            input_type = 'color'
-
-        class FieldTest(StringField):
-            widget = InputTest()
-
         class ModelTest(self.base):
             __tablename__ = 'model_test'
             query = None

--- a/tests/test_query_select_field.py
+++ b/tests/test_query_select_field.py
@@ -24,7 +24,7 @@ class LazySelect(object):
     def __call__(self, field, **kwargs):
         return list(
             (val, str(label), selected)
-            for val, label, selected in field.iter_choices()
+            for val, label, selected, _ in field.iter_choices()
         )
 
 

--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{39,310,311,312,313}-sqlalchemy{1.4,2.0}
+envlist = py{39,310,311,312,313}-sqlalchemy{1.4,2.0}-wtforms{3.1,3.2}
 
 [testenv]
 deps =
@@ -7,6 +7,8 @@ deps =
     pytest-cov
     sqlalchemy1.4: SQLAlchemy~=1.4
     sqlalchemy2.0: SQLAlchemy~=2.0
+    wtforms3.1: WTForms~=3.1
+    wtforms3.2: WTForms~=3.2
 commands = pip install -e ".[test]"
            py.test
 install_command = pip install {packages}

--- a/wtforms_alchemy/fields.py
+++ b/wtforms_alchemy/fields.py
@@ -214,10 +214,10 @@ class QuerySelectField(SelectFieldBase):
 
     def iter_choices(self):
         if self.allow_blank:
-            yield ('__None', self.blank_text, self.data is None)
+            yield ('__None', self.blank_text, self.data is None, {})
 
         for pk, obj in self._get_object_list():
-            yield (pk, self.get_label(obj), obj == self.data)
+            yield (pk, self.get_label(obj), obj == self.data, {})
 
     def process_formdata(self, valuelist):
         if valuelist:
@@ -289,7 +289,7 @@ class QuerySelectMultipleField(QuerySelectField):
 
     def iter_choices(self):
         for pk, obj in self._get_object_list():
-            yield (pk, self.get_label(obj), obj in self.data)
+            yield (pk, self.get_label(obj), obj in self.data, {})
 
     def process_formdata(self, valuelist):
         self._formdata = set(valuelist)
@@ -416,7 +416,8 @@ class GroupedQuerySelectField(SelectField):
                 (
                     self.coerce,
                     self.get_pk(self.data) if self.data else self.blank_value
-                )
+                ),
+                {}
             )
 
     def process_formdata(self, valuelist):
@@ -558,7 +559,8 @@ class GroupedQuerySelectMultipleField(SelectField):
                 (
                     self.coerce,
                     [self.get_pk(obj) for obj in self.data or []]
-                )
+                ),
+                {}
             )
 
     def process_formdata(self, valuelist):

